bson/codec_options.py

```diff
@@ -245,87 +245,92 @@ class _BaseCodecOptions(NamedTuple):
+
+        See :doc:`/examples/datetimes` for examples using the `tz_aware` and
+        `tzinfo` options.
+
+        See :doc:`/examples/uuid` for examples using the `uuid_representation`
+        option.
+
+        :Parameters:
+          - `document_class`: BSON documents returned in queries will be decoded
+            to an instance of this class. Must be a subclass of
+            :class:`~collections.abc.MutableMapping`. Defaults to :class:`dict`.
+          - `tz_aware`: If ``True``, BSON datetimes will be decoded to timezone
+            aware instances of :class:`~datetime.datetime`. Otherwise they will be
+            naive. Defaults to ``False``.
+          - `uuid_representation`: The BSON representation to use when encoding
+            and decoding instances of :class:`~uuid.UUID`. Defaults to
+            :data:`~bson.binary.UuidRepresentation.UNSPECIFIED`. New
+            applications should consider setting this to
+            :data:`~bson.binary.UuidRepresentation.STANDARD` for cross language
+            compatibility. See :ref:`handling-uuid-data-example` for details.
+          - `unicode_decode_error_handler`: The error handler to apply when
+            a Unicode-related error occurs during BSON decoding that would
+            otherwise raise :exc:`UnicodeDecodeError`. Valid options include
+            'strict', 'replace', 'backslashreplace', 'surrogateescape', and
+            'ignore'. Defaults to 'strict'.
+          - `tzinfo`: A :class:`~datetime.tzinfo` subclass that specifies the
+            timezone to/from which :class:`~datetime.datetime` objects should be
+            encoded/decoded.
+          - `type_registry`: Instance of :class:`TypeRegistry` used to customize
+            encoding and decoding behavior.
+          - `datetime_conversion`: Specifies how UTC datetimes should be decoded
+            within BSON. Valid options include 'datetime_ms' to return as a
+            DatetimeMS, 'datetime' to return as a datetime.datetime and
+            raising a ValueError for out-of-range values, 'datetime_auto' to
+            return DatetimeMS objects when the underlying datetime is
+            out-of-range and 'datetime_clamp' to clamp to the minimum and
+            maximum possible datetimes. Defaults to 'datetime'.
+
+        .. versionchanged:: 4.0
+           The default for `uuid_representation` was changed from
+           :const:`~bson.binary.UuidRepresentation.PYTHON_LEGACY` to
+           :const:`~bson.binary.UuidRepresentation.UNSPECIFIED`.
+
+        .. versionadded:: 3.8
+           `type_registry` attribute.
+
+        .. warning:: Care must be taken when changing
+           `unicode_decode_error_handler` from its default value ('strict').
+           The 'replace' and 'ignore' modes should not be used when documents
+           retrieved from the server will be modified in the client application
+           and stored back to the server.
+        """
+        return super().__init__()
```

Comment:
Don't we have to pass `*args, **kwargs` here?

Reply:
Ah, no, now I remember.  `namedtuple.__init__()` "takes exactly one argument (the instance to initialize))".  The args and kwargs are handled in `__new__()`.

Reply:
Ah okay. That's not intuitive at all :)